pkgs/by-name/ka/kak-tree-sitter-unwrapped/package.nix

```
@@ -0,0 +1,35 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromSourcehut,
+  nix-update-script,
+  testers,
+  kak-tree-sitter-unwrapped,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "kak-tree-sitter-unwrapped";
+  version = "1.1.2";
+
+  src = fetchFromSourcehut {
+    owner = "~hadronized";
+    repo = "kak-tree-sitter";
+    rev = "kak-tree-sitter-v${version}";
+    hash = "sha256-wBWfSyR8LGtug/mCD0bJ4lbdN3trIA/03AnCxZoEOSA=";
+  };
+
+  cargoHash = "sha256-v0DNcWPoHdquOlyPoPLoFulz66yCPR1W1Z3uuTjli5k=";
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion { package = kak-tree-sitter-unwrapped; };
+  };
```

Another idea:
```suggestion
    wrapper = callPackage ./wrapper.nix { };
  };
```

in which the wrapper has the same code you implemented as kak-tree-sitter-wrapped.

This way all files related to kak-tree-sitter are in the same directory.

Finally you can create an alias in all-packages.nix:

```nix
kak-tree-sitter-wrapped = kak-tree-sitter.wrapper;
```

---

On the other hand, I think it is still interesting to write the wrapper in the passthru.